### PR TITLE
integrating new stuff into Nevergrad

### PR DESCRIPTION
**Pull request by teytaud**
PR on a PR, for introducing LaMCTS into Nevergrad

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/optimization/lamcts/MCTS.py [hunk 1]
```diff
@@ -248,19 +254,46 @@ def search(self):
...
+    # f = Rosenrock(dims = 20)
+    # f = Schwefel(dims = 7)
+    # f = Zakharov()
+    # f = PushReward()
+    # f = Lunarlanding()
+    # f = Ant()
+    # f = HalfCheetah()
+    # f = Walker2d()
+    # f = Swimmer()
+    # f = Hopper()
```
Comment (jrapin) [inline nevergrad/optimization/lamcts/MCTS.py:296]: what is all this?

File: nevergrad/optimization/lamcts/MCTS.py [hunk 2]
```diff
@@ -30,10 +30,16 @@ def __init__(self, lb, ub, dims, ninits, func):
         self.samples                 =  []
         self.nodes                   =  []
         self.Cp                      =  5
-        self.lb                      =  lb
-        self.ub                      =  ub
+        assert (lb is None) == (ub is None), "We will deal with one-sided bounds later."
```
Comment (jrapin) [inline nevergrad/optimization/lamcts/MCTS.py:33]:
```suggestion
        assert (lb is None) == (ub is None), "One-sided bounds are not yet supported."
```

File: nevergrad/optimization/lamcts/MCTS.py [hunk 3]
```diff
@@ -30,10 +30,16 @@ def __init__(self, lb, ub, dims, ninits, func):
...
+        self.lb                      =  lb if lb is not None else -(np.pi/2)*np.ones(dims)
+        self.ub                      =  ub if ub is not None else (np.pi/2)*np.ones(dim)
         self.ninits                  =  ninits
-        self.func                    =  func
+        def nevergrad_func(x):
+            y = np.tan(x)
+            return func(y)
```
Comment (jrapin) [inline nevergrad/optimization/lamcts/MCTS.py:39]: what is this for?
Also, avoid inline functions, that would prevent pickling.

File: nevergrad/optimization/lamcts/MCTS.py (hunk 1 above)
Comment (teytaud) [reply nevergrad/optimization/lamcts/MCTS.py:296]: Test functions from the original LaMCTS code.

File: nevergrad/optimization/lamcts/MCTS.py (hunk 3 above)
Comment (teytaud) [reply nevergrad/optimization/lamcts/MCTS.py:39]: LaMCTS is not prepared for unbounded cases.

File: requirements/main.txt [hunk 4]
```diff
@@ -2,3 +2,8 @@ numpy>=1.15.0
 cma>=2.6.0
 bayesian-optimization>=1.1.0
 typing_extensions>=3.6.6
+torch
+gpytorch
+pygame
+gym
+matplotlib
```
Comment (jrapin) [inline requirements/main.txt:9]: those requirements should *not* be included by the default install, they are definitely too complex for just one case (and probably most of them should actually be in bench.txt)

Please notify me if you want to move forward with this PR

**Pull request merged**